Packs/ParseHTMLTables/Scripts/ParseHTMLTables/ParseHTMLTables.py

```diff
@@ -0,0 +1,235 @@
+    return ancestor is not None and node is not None and any([ancestor is p for p in node.parents])
+
+
+def parse_table(base: Optional[Union[BeautifulSoup, Tag, NavigableString]],
+                table_node: Union[BeautifulSoup, Tag, NavigableString]) -> Generator[Table, None, None]:
+    table = Table(title=find_table_title(base, table_node) or 'No Title')
+    has_nested_tables = False
+
+    node = table_node.find(['table', 'tr'])
+    while node and is_descendant(table_node, node):
+        if node.name == 'tr':
+            ths = [col.strip() for col in list_text(node, 'th')]
+            tds = [col.strip() for col in list_text(node, 'td')]
+            if tds:
+                table.add_row(columns=tds, labels=ths)
+            if ths and not table.get_header_labels():
+                table.set_header_labels(ths)
+
+            node = node.find_next(['table', 'tr'])
+
+        elif node.name == 'table':
+            has_nested_tables = True
+            yield from parse_table(base, node)
+
+            base = node.previous_element
+            node = node.find_next_sibling(True)
+        else:
+            node = node.find_next(['table', 'tr'])
+
+    # Not to make a table if tr only has tables
+    has_table = True
+    if has_nested_tables:
+        rows = table.get_rows()
+        if len(rows) == 1:
+            labels, cols = rows[0]
+            if len(cols) == 1 and not cols[0]:
+                has_table = False
+
+    if has_table:
+        yield table
+
+
+def parse_tables(node: Union[BeautifulSoup, Tag, NavigableString]) -> Generator[Table, None, None]:
+    base = None
+    node = node.find('table')
+    while node:
+        yield from parse_table(base, node)
+        base = node.next_sibling
+
+        while node:
+            next = node.find_next_sibling(True)
+            if next:
+                if next.name == 'table':
+                    break
+
+                next = next.find_next('table')
+                if next:
+                    break
+            node = node.parent
+        node = next
+
+
+def main():
+    args = demisto.args()
+    html = args.get('value') or ''
+    overwriting_title = args.get('title')
+    filter_indexes = argToList(args.get('filter_indexes'))
+    filter_titles = argToList(args.get('filter_titles'))
+
+    tables = []
+    soup = BeautifulSoup(html, 'html.parser')
+    index = -1
+    for table in parse_tables(soup):
+        rows = table.make_pretty_table_rows()
+        if not rows:
+            continue
+
+        index = index + 1
+        if filter_indexes and\
+           index not in filter_indexes and\
+           str(index) not in filter_indexes:
+            continue
+
+        original_title = table.get_title()
+        if filter_titles and original_title not in filter_titles:
+            continue
+
+        tables.append({overwriting_title or original_title: rows})
+
+    demisto.results(tables)
```

Comment:
Please handle exceptions in the main function. You can use the return_error function and add a proper error message and the specific error that was raised.